spec/API_specification/array_api/array_object.py

```diff
@@ -224,19 +224,40 @@ def __array_namespace__(self: array, /, *, api_version: Optional[str] = None) ->
 
     def __bool__(self: array, /) -> bool:
         """
-        Converts a zero-dimensional boolean array to a Python ``bool`` object.
+        Converts a zero-dimensional array to a Python ``bool`` object.
+
+        .. note::
+           When casting a real-valued input array to ``bool``, a value of ``0`` must cast to ``False``, and a non-zero value must cast to ``True``.
```

Comment:
Should note here whether inf and nan are included in this.

Reply:
You could say that the result should be equivalent to `value != 0`.  That is true for *all* supported types.

Reply:
Updated the note to say inputs equal/not-equal to `0`, which feels more correct anywho, i.e. should infer `-0.`, `0.+0.j`, etc. result in `False`.

<details>
<summary>Alternative solution using special cases</summary>

```python
    """
    ...
        **Special cases**

        For real-valued floating-point operands,

        - If ``self`` is ``NaN``, the result is ``True``.
        - If ``self`` is either ``+infinity`` or ``-infinity``, the result is ``True``.
        - If ``self`` is either ``+0`` or ``-0``, the result is ``False``.

        For complex floating-point operands, special cases must be handled as if the operation is implemented as ``bool(real(self)) and bool(imag(self))``.
    ...
    """
```

</details>

Reply:
Upon looking over this PR again, I think the proposed alternative solution which documents special cases should be preferred over using notes. This would ensure consistency with element-wise special cases found elsewhere in the spec.

Reply:
This discussion has been resolved for complex numbers and follows precedent elsewhere in the spec (e.g., `any` and [`all`](https://github.com/data-apis/array-api/blob/dff1997a5b4c455fb5b7be731780ce409d6e9a7d/spec/API_specification/array_api/utility_functions.py#L11)).